hatch_build.py

```diff
@@ -846,6 +846,17 @@ def initialize(self, version: str, build_data: dict[str, Any]) -> None:
         # field in core.metadata until this is possible
         self.metadata.core._optional_dependencies = self.optional_dependencies
 
+        # Add entrypoints dynamically for all provider packages, else they will not be found by
```

Comment:
Looks cool. The only thing is that `entrypoints` should be defined in `dynamic` array in pyproject.toml, otherwise, frontends will not know that they are available (pip might still read them regardless but technically speaking it is required:

https://peps.python.org/pep-0621/#dynamic

> If the metadata does not list a field in dynamic, then a build back-end CANNOT fill in the requisite metadata on behalf of the user (i.e. dynamic is the only way to allow a tool to fill in metadata and the user must opt into the filling in).

Reply:
Thanks to the links to specs. Needed to read multiple times to understand it. Makes sense... hope with "entry-points" it is the right key, was not sure if it would need to be named "project.entry-points" or "entry-points.'airflow.plugins'" or so. WDYT?

Reply:
Unfortunately, that's the nature of PEP that they are formal specification that is often result of a long discussion and what is captured there might need a bit more detailed reading. Enough said that 'packaging` is generally considered the most cofusing part of Python (and for good reasons I think) - I've heard that statement multiple times at PyCon.

Actually, I looked in more detail (now after PyCon I am back with a little Jet-Lag) and I think that even if they are defined in `core.metadata` it's not really the case. The `entrypoints` do not seem to be part of core-metadata, they are separate thing. 

You can see it for example here: https://pypi.org/project/apache-airflow/#files  -> download .whl file and you will see METADATA and entry_points.txt - so my suggestion above was actually wrong (sorry for that).

Instead what you should do, you should wrap this whole section in hatch_build in (see some of the sections above):

```python
if version != "standard":
```

The `version` there might be either `standard` (when .whl is built) or `editable` (when you install airflow in editable  and we definitely do not want airflow to produce "plugin" entrypoints that are usually produced by providers in .whl file - we really only want to have them when airflow is installed in editable mode.

Reply:
Also see here:

Reply:
Also see here: 

https://packaging.python.org/en/latest/specifications/pyproject-toml/#entry-points

Unlike dependencies and optional depenendencies, the entrypoints do not have corresponding "Core Metadata" fields (and to be a bit more confusing - pyproject.toml might specify entrypoints in various ways and they are sometimes conflicting. Also the `pyproject.toml`  entries are NOT the same as "Core Metadata" entries (for example `dependencies/optional-dependencies` in `pyproject.toml' are mapped to `[Requires-Dist](https://packaging.python.org/en/latest/specifications/core-metadata/#core-metadata-requires-dist) and [Provides-Extra](https://packaging.python.org/en/latest/specifications/core-metadata/#core-metadata-provides-extra) core metadata. 

It's not that it's wrong - It's all there but sometimes requires going through a series of docs and specs, so it's not intuitive and might be confusing. I got confused that in hatchlib's internal hook, entrypoints are stored in `core.metadata` simply - even if they are not part of it.

Reply:
Okay, then... another attempt pushed :-D

Correctly understood to revert the `dynamic`entry in pyproject.toml then?

Reply:
yes